src/analytics/from-redux.js

```diff
@@ -0,0 +1,41 @@
+/**
+ * WordPress dependencies
+ */
+import { use } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import eventTrackers from './trackers';
+import { REDUX_TRACKING } from './events';
+
+/**
+ * Initialize the Redux middleware
+ */
+export default function() {
+	use( ( registry ) => ( {
+		dispatch: ( namespace ) => {
+			const namespaceName =
+				typeof namespace === 'object' ? namespace.name : namespace;
+			const actions = { ...registry.dispatch( namespaceName ) };
+			const trackedEvents = REDUX_TRACKING[ namespaceName ];
+
+			if ( trackedEvents ) {
+				trackedEvents.forEach( ( actionName ) => {
+					const originalAction = actions[ actionName ];
+					const tracker = eventTrackers[ actionName ] || ( () => {} );
```

Comment:
I'm initializing the tracker to a noop function in case event is tracked but the tracker is not set up. The `try` should cover this case so we might want the error to show up so we know  the event is not fully tracked.

Reply:
As you said, because of the `try`, I would think a no-op is unnecessary. I would vote we remove the unnecessary no-op. 
```suggestion
					const tracker = eventTrackers[ actionName ];
```

Reply:
Feels like this is a case where it will only crash if we don't code this properly (and then it will always crash), so I'm on board with dropping the no-op. Let's go that way for now, but I'm happy to reconsider that decision later.